identity-service/src/notifications/notificationQueue.js

```diff
@@ -134,24 +138,73 @@ async function _sendNotification(notifFn, bufferObj, logger) {
  * Same as Promise.all(items.map(item => task(item))), but it waits for
  * the first {batchSize} promises to finish before starting the next batch.
  */
-async function promiseAllInBatches(task, logger, items, batchSize) {
+async function promiseAllInBatches(
+  task,
+  optimizelyClient,
+  logger,
+  items,
+  batchSize
+) {
   let position = 0
   let results = []
   while (position < items.length) {
     const itemsForBatch = items.slice(position, position + batchSize)
     results = [
       ...results,
       ...(await Promise.allSettled(
-        itemsForBatch.map((item) => task(logger, item))
+        itemsForBatch.map((item) => task(optimizelyClient, logger, item))
       ))
     ]
     position += batchSize
   }
   return results
 }
 
-async function processNotification(logger, notification) {
+const notificaitonTypeMapping = {
+  announcement: MappingVariable.PushAnnouncement,
+  [notificationTypes.Follow]: MappingVariable.PushFollow,
+  [notificationTypes.Repost.base]: MappingVariable.PushRepost,
+  [notificationTypes.Favorite.base]: MappingVariable.PushFavorite,
```

Comment:
why do these notif types need the .base?

Reply:
just legacy reasons of how we pass it along - in identity there's base which is only 'favorite' and then there's specifically 'favoriteTrack', 'favoritePlaylist', and 'favoriteAlbum'.... and we want to target all of them with the base. 
Bad design in identity